Native run of liquibase-quickstart fails with java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()

### Describe the bug

Native run of liquibase-quickstart fails with java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()

```
[INFO] Running org.acme.liquibase.MigrationEndpointIT
2023-05-08 21:54:43,118 WARN  [io.qua.hib.orm.dep.HibernateOrmProcessor] (build-10) Hibernate ORM is disabled because no JPA entities were found
2023-05-08 21:54:43,319 INFO  [io.qua.dev.h2.dep.H2DevServicesProcessor] (build-27) Dev Services for H2 started.
2023-05-08 21:54:43,322 INFO  [io.qua.dat.dep.dev.DevServicesDatasourceProcessor] (build-27) Dev Services for the default datasource (h2) started
Executing "/Users/rsvoboda/git/quarkus-quickstarts/liquibase-quickstart/target/liquibase-quickstart-1.0.0-SNAPSHOT-runner -Dquarkus.http.port=8081 -Dquarkus.http.ssl-port=8444 -Dtest.url=http://localhost:8081 -Dquarkus.log.file.path=/Users/rsvoboda/git/quarkus-quickstarts/liquibase-quickstart/target/quarkus.log -Dquarkus.log.file.enable=true -Dquarkus.log.category."io.quarkus".level=INFO -Dquarkus.datasource.password=quarkus -Dquarkus.datasource.db-kind=h2 -Dquarkus.datasource.jdbc.url=jdbc:h2:tcp://localhost:58844/mem:quarkus;DB_CLOSE_DELAY=-1 -Dquarkus.datasource.username=quarkus"
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0, Time elapsed: 4.426 s <<< FAILURE! - in org.acme.liquibase.MigrationEndpointIT
[ERROR] org.acme.liquibase.MigrationEndpointIT.testListAllChangeStatus  Time elapsed: 1.459 s  <<< FAILURE!
java.lang.AssertionError:
1 expectation failed.
Expected status code <200> but was <500>.
...
[ERROR] Failures:
[ERROR]   MigrationEndpointIT>MigrationEndpointTest.testListAllChangeStatus:17 1 expectation failed.
Expected status code <200> but was <500>.

[INFO]
[ERROR] Tests run: 1, Failures: 1, Errors: 0, Skipped: 0
[INFO]
[WARNING] Corrupted channel by directly writing to native stream in forked JVM 1. See FAQ web page and the dump file /Users/rsvoboda/git/quarkus-quickstarts/liquibase-quickstart/target/failsafe-reports/2023-05-08T21-54-41_060-jvmRun1.dumpstream
```

Content of 2023-05-08T21-54-41_060-jvmRun1.dumpstream, contains strange `Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.UnexpectedLiquibaseException: java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()'` as if the native build over-optimized the native binary. JVM mode passes the tests, and migration succeeds.
```
# Created at 2023-05-08T21:54:44.143
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '__  ____  __  _____   ___  __ ____  ______ '.

# Created at 2023-05-08T21:54:44.144
Corrupted channel by directly writing to native stream in forked JVM 1. Stream ' --/ __ \/ / / / _ | / _ \/ //_/ / / / __/ '.

# Created at 2023-05-08T21:54:44.145
Corrupted channel by directly writing to native stream in forked JVM 1. Stream ' -/ /_/ / /_/ / __ |/ , _/ ,< / /_/ /\ \   '.

# Created at 2023-05-08T21:54:44.145
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '--\___\_\____/_/ |_/_/|_/_/|_|\____/___/   '.

# Created at 2023-05-08T21:54:44.165
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,141 INFO  [liq.database] (main) Set default schema name to PUBLIC'.

# Created at 2023-05-08T21:54:44.167
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,164 INFO  [liq.lockservice] (main) Successfully acquired change log lock'.

# Created at 2023-05-08T21:54:44.168
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,167 INFO  [liq.parser] (main) INFO: An older version of the XSD is specified in one or more changelog's <databaseChangeLog> header. This can lead to unexpected outcomes. If a specific XSD is not required, please replace all XSD version references with "-latest". Learn more at https://docs.liquibase.com'.

# Created at 2023-05-08T21:54:44.194
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'INFO: An older version of the XSD is specified in one or more changelog's <databaseChangeLog> header. This can lead to unexpected outcomes. If a specific XSD is not required, please replace all XSD version references with "-latest". Learn more at https://docs.liquibase.com'.

# Created at 2023-05-08T21:54:44.200
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,193 INFO  [liquibase] (main) Parsed changelog file 'db/changeLog.xml''.

# Created at 2023-05-08T21:54:44.215
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,199 INFO  [liq.servicelocator] (main) Cannot load service: liquibase.command.CommandStep: Provider liquibase.command.core.StartH2CommandStep not found'.

# Created at 2023-05-08T21:54:44.220
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,214 INFO  [liq.changelog] (main) Creating database history table with name: PUBLIC.DATABASECHANGELOG'.

# Created at 2023-05-08T21:54:44.221
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,219 INFO  [liq.command] (main) Using deploymentId: 3575684218'.

# Created at 2023-05-08T21:54:44.223
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,220 INFO  [liq.changelog] (main) Reading from PUBLIC.DATABASECHANGELOG'.

# Created at 2023-05-08T21:54:44.224
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Running Changeset: db/changeLog.xml::1::quarkus'.

# Created at 2023-05-08T21:54:44.225
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,224 INFO  [liq.changelog] (main) Table quarkus created'.

# Created at 2023-05-08T21:54:44.228
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,224 INFO  [liq.changelog] (main) ChangeSet db/changeLog.xml::1::quarkus ran successfully in 2ms'.

# Created at 2023-05-08T21:54:44.229
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) UPDATE SUMMARY'.

# Created at 2023-05-08T21:54:44.230
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) Run:                          1'.

# Created at 2023-05-08T21:54:44.230
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) Previously run:               0'.

# Created at 2023-05-08T21:54:44.231
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) Filtered out:                 0'.

# Created at 2023-05-08T21:54:44.232
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) -------------------------------'.

# Created at 2023-05-08T21:54:44.232
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) Total change sets:            1'.

# Created at 2023-05-08T21:54:44.233
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'UPDATE SUMMARY'.

# Created at 2023-05-08T21:54:44.233
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Run:                          1'.

# Created at 2023-05-08T21:54:44.234
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Previously run:               0'.

# Created at 2023-05-08T21:54:44.234
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Filtered out:                 0'.

# Created at 2023-05-08T21:54:44.235
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '-------------------------------'.

# Created at 2023-05-08T21:54:44.235
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Total change sets:            1'.

# Created at 2023-05-08T21:54:44.236
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.util] (main) Update summary generated'.

# Created at 2023-05-08T21:54:44.236
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,228 INFO  [liq.command] (main) Update command completed successfully.'.

# Created at 2023-05-08T21:54:44.237
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Liquibase: Update has been successful.'.

# Created at 2023-05-08T21:54:44.237
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,230 INFO  [liq.lockservice] (main) Successfully released change log lock'.

# Created at 2023-05-08T21:54:44.238
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,233 INFO  [liq.lockservice] (main) Successfully released change log lock'.

# Created at 2023-05-08T21:54:44.238
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,233 INFO  [liq.command] (main) Command execution complete'.

# Created at 2023-05-08T21:54:44.249
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,235 INFO  [liq.lockservice] (main) Successfully released change log lock'.

# Created at 2023-05-08T21:54:44.250
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,249 INFO  [io.quarkus] (main) liquibase-quickstart 1.0.0-SNAPSHOT native (powered by Quarkus 999-SNAPSHOT) started in 0.155s. Listening on: http://0.0.0.0:8081'.

# Created at 2023-05-08T21:54:44.251
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,249 INFO  [io.quarkus] (main) Profile prod activated. '.

# Created at 2023-05-08T21:54:44.663
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:44,249 INFO  [io.quarkus] (main) Installed features: [agroal, cdi, hibernate-orm, jdbc-h2, liquibase, narayana-jta, resteasy-reactive, resteasy-reactive-jackson, smallrye-context-propagation, vertx]'.

# Created at 2023-05-08T21:54:45.545
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,539 INFO  [liq.database] (executor-thread-1) Set default schema name to PUBLIC'.

# Created at 2023-05-08T21:54:45.546
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,545 INFO  [liq.lockservice] (executor-thread-1) Successfully acquired change log lock'.

# Created at 2023-05-08T21:54:45.551
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,545 INFO  [liq.command] (executor-thread-1) Dropping Database Objects in schema: QUARKUS.PUBLIC'.

# Created at 2023-05-08T21:54:45.557
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,550 INFO  [liq.lockservice] (executor-thread-1) Successfully released change log lock'.

# Created at 2023-05-08T21:54:45.558
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,557 INFO  [liq.command] (executor-thread-1) Command execution complete'.

# Created at 2023-05-08T21:54:45.559
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,558 ERROR [io.qua.ver.htt.run.QuarkusErrorHandler] (executor-thread-1) HTTP Request to /migration/status failed, error id: 2db66e3e-0732-43d3-af13-4f1fd6381566-1: liquibase.exception.DatabaseException: liquibase.exception.CommandExecutionException: liquibase.exception.UnexpectedLiquibaseException: java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()'.

# Created at 2023-05-08T21:54:45.560
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.Liquibase.dropAll(Liquibase.java:1154)'.

# Created at 2023-05-08T21:54:45.560
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.Liquibase.dropAll(Liquibase.java:1131)'.

# Created at 2023-05-08T21:54:45.560
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.acme.liquibase.MigrationService.checkMigration(MigrationService.java:19)'.

# Created at 2023-05-08T21:54:45.561
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.acme.liquibase.MigrationService_ClientProxy.checkMigration(Unknown Source)'.

# Created at 2023-05-08T21:54:45.561
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.acme.liquibase.MigrationResource.migrationStatus(MigrationResource.java:26)'.

# Created at 2023-05-08T21:54:45.562
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.acme.liquibase.MigrationResource_ClientProxy.migrationStatus(Unknown Source)'.

# Created at 2023-05-08T21:54:45.562
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.acme.liquibase.MigrationResource$quarkusrestinvoker$migrationStatus_6b8b5a4a70019705845adafdf9fba90236c27b0c.invoke(Unknown Source)'.

# Created at 2023-05-08T21:54:45.563
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.jboss.resteasy.reactive.server.handlers.InvocationHandler.handle(InvocationHandler.java:29)'.

# Created at 2023-05-08T21:54:45.564
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at io.quarkus.resteasy.reactive.server.runtime.QuarkusResteasyReactiveRequestContext.invokeHandler(QuarkusResteasyReactiveRequestContext.java:141)'.

# Created at 2023-05-08T21:54:45.564
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.jboss.resteasy.reactive.common.core.AbstractResteasyReactiveContext.run(AbstractResteasyReactiveContext.java:145)'.

# Created at 2023-05-08T21:54:45.565
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at io.quarkus.vertx.core.runtime.VertxCoreRecorder$14.runWith(VertxCoreRecorder.java:576)'.

# Created at 2023-05-08T21:54:45.565
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.jboss.threads.EnhancedQueueExecutor$Task.run(EnhancedQueueExecutor.java:2513)'.

# Created at 2023-05-08T21:54:45.566
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.jboss.threads.EnhancedQueueExecutor$ThreadBody.run(EnhancedQueueExecutor.java:1538)'.

# Created at 2023-05-08T21:54:45.566
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.jboss.threads.DelegatingRunnable.run(DelegatingRunnable.java:29)'.

# Created at 2023-05-08T21:54:45.567
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.jboss.threads.ThreadLocalResettingRunnable.run(ThreadLocalResettingRunnable.java:29)'.

# Created at 2023-05-08T21:54:45.568
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at io.netty.util.concurrent.FastThreadLocalRunnable.run(FastThreadLocalRunnable.java:30)'.

# Created at 2023-05-08T21:54:45.568
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at java.base@17.0.6/java.lang.Thread.run(Thread.java:833)'.

# Created at 2023-05-08T21:54:45.569
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.graalvm.nativeimage.builder/com.oracle.svm.core.thread.PlatformThreads.threadStartRoutine(PlatformThreads.java:775)'.

# Created at 2023-05-08T21:54:45.570
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at org.graalvm.nativeimage.builder/com.oracle.svm.core.posix.thread.PosixPlatformThreads.pthreadStartRoutine(PosixPlatformThreads.java:203)'.

# Created at 2023-05-08T21:54:45.570
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Caused by: liquibase.exception.CommandExecutionException: liquibase.exception.UnexpectedLiquibaseException: java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()'.

# Created at 2023-05-08T21:54:45.571
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.command.CommandScope.execute(CommandScope.java:235)'.

# Created at 2023-05-08T21:54:45.571
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.Liquibase.dropAll(Liquibase.java:1152)'.

# Created at 2023-05-08T21:54:45.572
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	... 18 more'.

# Created at 2023-05-08T21:54:45.572
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Caused by: liquibase.exception.UnexpectedLiquibaseException: java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()'.

# Created at 2023-05-08T21:54:45.573
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:334)'.

# Created at 2023-05-08T21:54:45.573
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:435)'.

# Created at 2023-05-08T21:54:45.574
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.includeNestedObjects(DatabaseSnapshot.java:369)'.

# Created at 2023-05-08T21:54:45.574
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:332)'.

# Created at 2023-05-08T21:54:45.575
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.init(DatabaseSnapshot.java:106)'.

# Created at 2023-05-08T21:54:45.575
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.<init>(DatabaseSnapshot.java:59)'.

# Created at 2023-05-08T21:54:45.576
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.JdbcDatabaseSnapshot.<init>(JdbcDatabaseSnapshot.java:34)'.

# Created at 2023-05-08T21:54:45.576
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:214)'.

# Created at 2023-05-08T21:54:45.577
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:243)'.

# Created at 2023-05-08T21:54:45.577
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.SnapshotGeneratorFactory.createSnapshot(SnapshotGeneratorFactory.java:227)'.

# Created at 2023-05-08T21:54:45.578
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.lockservice.StandardLockService.destroy(StandardLockService.java:496)'.

# Created at 2023-05-08T21:54:45.578
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.command.core.InternalDropAllCommandStep.run(InternalDropAllCommandStep.java:118)'.

# Created at 2023-05-08T21:54:45.579
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.command.CommandScope.execute(CommandScope.java:212)'.

# Created at 2023-05-08T21:54:45.579
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	... 19 more'.

# Created at 2023-05-08T21:54:45.579
Corrupted channel by directly writing to native stream in forked JVM 1. Stream 'Caused by: java.lang.NoSuchMethodException: java.util.concurrent.ConcurrentHashMap.<init>()'.

# Created at 2023-05-08T21:54:45.580
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at java.base@17.0.6/java.lang.Class.getConstructor0(DynamicHub.java:3585)'.

# Created at 2023-05-08T21:54:45.580
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at java.base@17.0.6/java.lang.Class.getConstructor(DynamicHub.java:2271)'.

# Created at 2023-05-08T21:54:45.581
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.replaceObject(DatabaseSnapshot.java:475)'.

# Created at 2023-05-08T21:54:45.581
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.includeNestedObjects(DatabaseSnapshot.java:369)'.

# Created at 2023-05-08T21:54:45.620
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	at liquibase.snapshot.DatabaseSnapshot.include(DatabaseSnapshot.java:332)'.

# Created at 2023-05-08T21:54:45.620
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '	... 31 more'.

# Created at 2023-05-08T21:54:46.125
Corrupted channel by directly writing to native stream in forked JVM 1. Stream '2023-05-08 21:54:45,619 INFO  [io.quarkus] (Shutdown thread) liquibase-quickstart stopped in 0.002s'.
```


### Expected behavior

Native run works


### Actual behavior

Native run fails

### How to Reproduce?

Run https://github.com/quarkusio/quarkus-quickstarts/tree/development/liquibase-quickstart with Quarkus main

### Output of `uname -a` or `ver`

macOS

### Output of `java -version`

openjdk version "17.0.6" 2023-01-17

### GraalVM version (if different from Java)

OpenJDK Runtime Environment GraalVM CE 22.3.1 (build 17.0.6+10-jvmci-22.3-b13)

### Quarkus version or git rev

main

### Build tool (ie. output of `mvnw --version` or `gradlew --version`)

_No response_

### Additional information

_No response_

--- a/extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java
+++ b/extensions/liquibase/deployment/src/main/java/io/quarkus/liquibase/deployment/LiquibaseProcessor.java
@@ -14,6 +14,7 @@ import java.util.List;
 import java.util.Map;
 import java.util.Optional;
 import java.util.Set;
+import java.util.concurrent.ConcurrentHashMap;
 import java.util.function.BiConsumer;
 import java.util.stream.Collectors;
 import java.util.stream.Stream;
@@ -151,6 +152,9 @@ class LiquibaseProcessor {
         reflective.produce(ReflectiveClassBuildItem.builder(
                 liquibase.change.ConstraintsConfig.class.getName())
                 .fields().build());
+
+        // liquibase seems to instantiate these types reflectively...
+        reflective.produce(ReflectiveClassBuildItem.builder(ConcurrentHashMap.class, ArrayList.class).build());
 
         // register classes marked with @DatabaseChangeProperty for reflection
         Set<String> classesMarkedWithDatabaseChangeProperty = new HashSet<>();

